kolibri/core/discovery/api.py

```diff
@@ -1,21 +1,32 @@
 from rest_framework import viewsets
 from rest_framework.response import Response
 
+from .models import DynamicNetworkLocation
 from .models import NetworkLocation
+from .models import StaticNetworkLocation
 from .serializers import NetworkLocationSerializer
 from kolibri.core.content.permissions import CanManageContent
-from kolibri.core.device.permissions import UserHasAnyDevicePermissions
-from kolibri.core.discovery.utils.network.search import get_available_instances
+from kolibri.core.discovery.utils.network.search import run_peer_discovery
 
 
 class NetworkLocationViewSet(viewsets.ModelViewSet):
     permission_classes = (CanManageContent,)
     serializer_class = NetworkLocationSerializer
-    queryset = NetworkLocation.objects.all()
+    queryset = NetworkLocation.objects
 
 
-class NetworkSearchViewSet(viewsets.ViewSet):
-    permission_classes = (UserHasAnyDevicePermissions,)
+class DynamicNetworkLocationViewSet(viewsets.ModelViewSet):
+    permission_classes = (CanManageContent,)
+    serializer_class = NetworkLocationSerializer
+    queryset = DynamicNetworkLocation.objects
 
     def list(self, request):
```

Comment:
This endpoint is the main interface to discovered locations.  It both runs peer discovery and serializes the current cache of `DynamicNetworkLocation` objects.  During serialization, it will be calling `DynamicNetworkLocation.available`, which will automatically delete unavailable locations.  So it also prunes unavailable locations as well.